docs/src/upgrade/release-3.4.x.asciidoc

```diff
@@ -29,6 +29,29 @@ Please see the link:https://github.com/apache/tinkerpop/blob/3.4.0/CHANGELOG.asc
 
 === Upgrading for Users
 
+==== Changed infix behavior
+
+The infix notation of `and()` and `or()` now supports an arbitrary number of traversals and `ConnectiveStrategy` produces a traversal with the correct AND and OR semantics. Furthermore,
+previous versions failed to apply 3 or more `and()` steps in an infix notation, this is now fixed.
+
```

Comment:
The graph example is good for those familiar with the graph but it's not immediately clear what the difference in behavior is. I really like the example from the comments. I think it would be good to add here (not necessarily verbatim) for clarity. 
```
# BEHAVIOR

Input: a.or.b.and.c.or.d.and.e.or.f.and.g.and.h.or.i

## BEFORE
Output: or(a, or(and(b, c), or(and(d, e), or(and(and(f, g), h), i))))

## NOW
Output: or(a, and(b, c), and(d, e), and(f, g, h), i)
```